hibernate-core/src/test/java/org/hibernate/test/schemaupdate/SchemaCreationToOutputScriptTest.java

```diff
@@ -0,0 +1,163 @@
+/*
+ * Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later.
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.test.schemaupdate;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.nio.file.Files;
+import java.util.HashMap;
+import java.util.List;
+import javax.persistence.Entity;
+import javax.persistence.Id;
+
+import org.hibernate.boot.MetadataSources;
+import org.hibernate.boot.registry.StandardServiceRegistryBuilder;
+import org.hibernate.boot.spi.MetadataImplementor;
+import org.hibernate.cfg.AvailableSettings;
+import org.hibernate.cfg.Environment;
+import org.hibernate.dialect.H2Dialect;
+import org.hibernate.engine.config.spi.ConfigurationService;
+import org.hibernate.service.ServiceRegistry;
+import org.hibernate.tool.schema.spi.SchemaManagementToolCoordinator;
+
+import org.hibernate.testing.RequiresDialect;
+import org.hibernate.testing.ServiceRegistryBuilder;
+import org.hibernate.testing.TestForIssue;
+import org.hibernate.testing.junit4.BaseUnitTestCase;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.Assert.assertTrue;
+
+@TestForIssue(jiraKey = "HHH-11817")
+@RequiresDialect(H2Dialect.class)
+public class SchemaCreationToOutputScriptTest extends BaseUnitTestCase {
+
+	private final String createTableMyEntity = "create table MyEntity";
+	private final String createTableMySecondEntity = "create table MySecondEntity";
+
+	private File output;
+	private ServiceRegistry serviceRegistry;
+	private MetadataImplementor metadata;
+
+	@Before
+	public void setUp() throws Exception {
+		output = File.createTempFile( "creation_script", ".sql" );
+		output.deleteOnExit();
+		try (FileWriter writer = new FileWriter( output )) {
```

Comment:
Technically speaking, we should avoid writing to file without buffering in legacy Java file IO for that is a big performance drain. But given this is a testing case, the concern might be discarded. Alternatively we can use Java NIO file processing without such concern at all:
```
public static Path write(Path path, byte[] bytes, OpenOption... options)
        throws IOException
```
see https://mkyong.com/java/how-to-write-data-to-temporary-file-in-java/

Reply:
Thanks @NathanQingyangXu , used the buffering!